// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.jcoin";
option java_outer_classname = "JCoinProto";
option objc_class_prefix = "JCN";

package jcoin;


service Registrar {
  rpc Register (NewNode) returns (LastNode) {}
}

message NewNode {
  string nVersion = 1;
  string nTime = 2;
  string addrMe = 3;
}

message LastNode {
  string lastNodeIp = 1;
}

service Handshake {
  rpc Shake (CallerShake) returns (ReceiverShake) {}
}

message CallerShake {
  string nVersion = 1;
  string nTime = 2;
  string addrMe = 3;
  int32 bestHeight = 4;
}

message ReceiverShake {
  repeated string knownPeers = 1;
}

service TransactionCast {

}

service BlockCast {

}
